code/modules/mob/living/carbon/brain/brain_item.dm

```diff
@@ -69,23 +68,36 @@
 	owner.brain_op_stage = 4.0
 
 	var/obj/item/organ/brain/B = src
-	if(istype(B) && istype(owner))
+	if(istype(B) && istype(owner) && owner.internal_organs_by_name[organ_tag] == src)

```

Comment:
Actually, I think the transfer identity proc will overwrite the brain's DNA to that of the body they were in. This will lead to... quite some weirdness, including that inserting a brain and removing it will change its genetic identity. Not quite sure how to address.